spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -548,8 +549,83 @@
     let(:role) { create(:student) }
 
     context 'GET student_interface' do
-      before { get_as role, :student_interface, params: params }
-      # TODO: write tests
+      let(:assignment) do
+        create(:assignment,
+               assignment_properties_attributes: {
+                 enable_student_tests: true,
+                 tokens_per_period: 5,
+                 token_start_date: 1.day.ago,
+                 token_period: 24
+               })
+      end
+      let(:grouping) { create(:grouping, assignment: assignment) }
+      let(:student) { create(:student) }
+      let(:role) { student }
+      let(:params) { { course_id: assignment.course.id, assignment_id: assignment.id } }
+
+      before do
+        create(:student_membership, role: student, grouping: grouping, membership_status: 'accepted')
+        sign_in student
+      end
+
+      context 'when student tests are enabled and the student has a grouping' do
+        before do
+          get_as student, :student_interface, params: params
+        end
+
+        it 'should calculate the next token generation time' do
+          assignment.reload
+          current_time = Time.zone.now.change(hour: 14)
+          token_start_time = assignment.token_start_date
+          hours_since_start = [(current_time - token_start_time) / 1.hour, 0].max
```

Comment:
It's good that this test is more specific now. However, because you've chosen specific numbers, you shouldn't just copy the code from the implementation, as that won't reveal bugs. Instead, you should separately figure out what time should be displayed, and use that in this test case.

In this case, based on the test configuration you've set up, if you travel to (e.g.) 14 hours in the future, the next token generation time should be 2 days from when the tokens started being available.